tests/test_sim_result.py

```diff
@@ -36,7 +38,245 @@ def test_pickle(self):
+        plot_test = event_states.plot() # Plot doesn't raise error
+
+    def test_not_implemented(self):
+        # Not implemented functions, should raise errors
+        NUM_ELEMENTS = 5
+        time = list(range(NUM_ELEMENTS))
+        state = [i * 2.5 for i in range(NUM_ELEMENTS)]
+        result = SimResult(time, state)
+        self.assertRaises(NotImplementedError, result.append)
+        self.assertRaises(NotImplementedError, result.count)
+        self.assertRaises(NotImplementedError, result.insert)
+        self.assertRaises(NotImplementedError, result.reverse)
+
+    # Tests for LazySimResult
+    def test_lazy_data_fcn(self):
+        def f(x):
+            return x * 2
+        NUM_ELEMENTS = 5
+        time = list(range(NUM_ELEMENTS))
+        state = [i * 2.5 for i in range(NUM_ELEMENTS)]
+        result = LazySimResult(f, time, state)
+        self.assertEqual(result.data, [0.0, 5.0, 10.0, 15.0, 20.0])
+
+    def test_lazy_clear(self):
+        def f(x):
+            return x * 2
+        NUM_ELEMENTS = 5
+        time = list(range(NUM_ELEMENTS))
+        state = [i * 2.5 for i in range(NUM_ELEMENTS)]
+        result = LazySimResult(f, time, state)
+        self.assertEqual(result.times, [0, 1, 2, 3, 4])
+        self.assertEqual(result.data, [0.0, 5.0, 10.0, 15.0, 20.0])
+        self.assertEqual(result.states, [0.0, 2.5, 5.0, 7.5, 10.0])
+        self.assertRaises(TypeError, result.clear, True)
+
+        result.clear()
+        self.assertEqual(result.times, [])
+        self.assertEqual(result.data, [])
+        self.assertEqual(result.states, [])
+
+    def test_lazy_extend(self):
+        def f(x):
+            return x * 2
+        NUM_ELEMENTS = 5
+        time = list(range(NUM_ELEMENTS))
+        state = [i * 2.5 for i in range(NUM_ELEMENTS)]
+        result = LazySimResult(f, time, state)
+
+        def f2(x):
+            return x * 5
+        NUM_ELEMENTS = 10
+        time2 = list(range(NUM_ELEMENTS))
+        state2 = [i * 5 for i in range(NUM_ELEMENTS)]
+        result2 = LazySimResult(f2, time2, state2)
+        self.assertEqual(result.times, [0, 1, 2, 3, 4]) # Assert data is correct before extending
+        self.assertEqual(result.data, [0.0, 5.0, 10.0, 15.0, 20.0])
+        self.assertEqual(result.states, [0.0, 2.5, 5.0, 7.5, 10.0])
+        self.assertEqual(result2.times, [0, 1, 2, 3, 4, 5, 6, 7, 8, 9])
+        self.assertEqual(result2.data, [0, 25, 50, 75, 100, 125, 150, 175, 200, 225])
+        self.assertEqual(result2.states, [0, 5, 10, 15, 20, 25, 30, 35, 40, 45])
+
+        result.extend(result2)
+        self.assertEqual(result.times, [0, 1, 2, 3, 4, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9]) # Assert data is correct after extending
+        self.assertEqual(result.data, [0.0, 5.0, 10.0, 15.0, 20.0, 0, 10, 20, 30, 40, 50, 60, 70, 80, 90])
+        self.assertEqual(result.states, [0.0, 2.5, 5.0, 7.5, 10.0, 0, 5, 10, 15, 20, 25, 30, 35, 40, 45])
+
```

Comment:
Another test to add. Extending should reset cache. 

So, if you do this:
   * Build lazy sim result (line 234)
   * Access data (lines 235-241)
   * Check is_cached (should be true here)
   * Extend (line 242)
   * Check is_cached (should be false here)
   * Access data (lines 243-245)